modules/nlu/src/backend/tools/stopWords.ts

```diff
@@ -0,0 +1,39 @@
+import { createReadStream } from 'fs'
+import fs from 'fs-extra'
+import path from 'path'
+import readline from 'readline'
+
+fs.exists
+
+const StopWordsByLang: _.Dictionary<string[]> = {}
+
+async function loadStopWords(language: string): Promise<string[]> {
+  const fn = path.join(__dirname, `stopwords/${language}.txt`)
+
+  const langSupported: boolean = await Promise.fromCallback(callback => {
+    fs.exists(fn, callback.bind(this, undefined))
+  })
+  if (!langSupported) {
+    return []
+  }
+
+  return new Promise((resolve, reject) => {
+    const stopWords = []
+    const stream = createReadStream(fn)
+    const rl = readline.createInterface({ input: stream, crlfDelay: Infinity })
+
+    rl.on('line', l => {
+      stopWords.push(l)
+    })
+
+    rl.on('close', () => resolve(stopWords))
```

Comment:
What do you think of converting your stopword.txt file to stopwords.json and feeding it an array of words ? You could then simply require('stopwords.json') instead of this stream

Reply:
does it really make a difference ? reading a file or parsing a json file are probably the same complexity.  If this really hurts so bad then I can do as you suggest

Reply:
Not really. Just less code to write, and easier format to handle. But if the stopwords file gets really big then your solution is better since it's async.